### DISCUSSION TITLE
Generating twin and multiple twinning in a grain

### DISCUSSION THREAD

**Discussion by PengWei97** [neperfepx/neper#697] [Q&A] [answered]
Dear Neper experts,

I am currently using Neper to generate microstructures with different features. I am interested in creating a microstructure that includes a twin or multiple twins within a grain. I have studied the #401 dialogue in the forum, and I have used the command 
`neper -T -dim 2 -n "10::msfile(n_msfile)" -morpho "gg::msfile(morpho_msfile)" -o d`
to generate a microstructure. However, the resulting microstructure is not what I expected, as shown in the attached figure.
![image](https://github.com/neperfepx/neper/assets/56183775/b1297cdf-6c1a-4ecb-be1e-250fc0504d4a)

I am wondering if anyone can provide me with guidance on how to introduce twins into my microstructure using Neper. I would greatly appreciate any advice or suggestions on how to accomplish this.

Thank you in advance for your help.

Sincerely,

wei

**Comment by mkasemer** [comment]
See: https://neper.info/doc/neper_t.html#cmdoption-morpho

You want to both specify a variable width in your width file, as well as a planar normal to control the spatial orientation.

**Reply by smallrain1996** [reply]
![image](https://github.com/neperfepx/neper/assets/52028117/960c2ff9-7643-444b-9b4c-e7624168c4f2)

**Reply by PengWei97** [reply, marked as answer]
Thank you so much for your help and suggestions regarding this question on how to introduce twinning in a microstructure generated using Neper. I am happy to inform you that I have successfully addressed this issue by configuring "lamellar(w=0.1:0.02:0.08:0.02:1.0,v=(1,-0.5,0),pos=start)" in the script.
![image](https://github.com/neperfepx/neper/assets/56183775/78082324-0d78-48c3-86e9-562915b9437d)

Once again, thank you for your invaluable help.
